### PR TITLE
Preliminary Selenium Test

### PR DESCRIPTION
**Pull request by borjainglefernandez**
## Status

- In progress

## GitHub Issues addressed

- This PR closes #453 

## What I did

- Wrote two preliminarys selenium test to test that it works
- Changed precommit.sh to run that test script with command line arguments

## Questions/Discussions/Notes
-Not running it in Docker lol

## Merging the PR

- Who should merge this PR?
  - [] I will merge it myself
  - [] The last reviewer to approve can merge it
- Should the commit history be preserved in the base branch, or is it okay to combine all of them into a single commit ("squash-merge")?
  - [] preserve the history
  - [] squash-merge

### CODE REVIEW

File: seleniumtests/selenium_test.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from selenium import webdriver
+from selenium.webdriver.chrome.options import Options
+from webdriver_manager.chrome import ChromeDriverManager
+import unittest
+
+
+class SeleniumTests(unittest.TestCase):
+    def setUp(self):
+        chrome_options = Options()
+        chrome_options.add_argument('--headless')
```
Comment (borjainglefernandez) [inline seleniumtests/selenium_test.py:10]: I forgot to comment this but headless mode essentially means it doesn't open chrome like it normally would and show exactly what it's doing.

**Comment by Bartleby2718** [comment]
In CS 4260 ISA, I used Selenium docker container to test Django containers for CS 4260 ISA.

Having skimmed through https://docs.djangoproject.com/en/3.1/topics/testing/tools/#liveservertestcase, I guess another way would be to use fixtures to include an admin account, log in to the admin page (in order to circumvent Google Sign In), and then go to the normal page and start working.

**Comment by alex-shen1** [comment]
Friendly reminder - if you're creating a PR that's still in progress, please attach the "In progress" label on the PRs page! Helps with organization/knowing which PRs need to be looked at.

File: precommit.sh [hunk 2]
```diff
@@ -8,3 +8,5 @@ docker exec tcf_django node_modules/.bin/eslint --cache --fix -c .config/.eslint
 docker exec tcf_django coverage run manage.py test --keepdb
 # Get code coverage.
 docker exec tcf_django coverage report -m
+
+python3 seleniumtests/selenium_test.py
```
Comment (alex-shen1) [inline precommit.sh:12]: I also know very little about Selenium, but generally I believe we want everything to be run in the Docker container (i.e. put `docker exec tcf_django` before all the commands). If running the tests outside Docker is is a deliberate choice on your part, please ignore!

**Comment by borjainglefernandez** [comment]
> Friendly reminder - if you're creating a PR that's still in progress, please attach the "In progress" label on the PRs page! Helps with organization/knowing which PRs need to be looked at.
> 
> EDIT: better yet, you can create PRs as "drafts". I believe this should still let everyone view/comment like normal, it'll just be explicitly delineated from the need-review PRs.

Will do. My only concern is I uploaded this as PR specifically to get feedback so it technically is a need-review PR.

File: precommit.sh (hunk 2 above)
Comment (borjainglefernandez) [reply precommit.sh:12]: Yes, I wrote about this in my concerns. Thank you for giving me the actual command to run this lol

File: precommit.sh [hunk 3]
```diff
@@ -8,3 +8,6 @@ docker exec tcf_django node_modules/.bin/eslint --cache --fix -c .config/.eslint
 docker exec tcf_django coverage run manage.py test --keepdb
 # Get code coverage.
 docker exec tcf_django coverage report -m
+# Selenium tests.
+python seleniumtests/browse_selenium_test.py $@
```
Comment (borjainglefernandez) [inline precommit.sh:12]: I tried and tried and tried to get it to work on docker but I just can't lol I'll revisit this later.

Comment (borjainglefernandez) [reply precommit.sh:12]: Basically the issue is if put `docker exec tcf_django` in front, the entire chrome driver fails

File: precommit.sh [hunk 4]
```diff
@@ -8,3 +8,6 @@ docker exec tcf_django node_modules/.bin/eslint --cache --fix -c .config/.eslint
...
+python seleniumtests/course_page_selenium_test.py $@
```
Comment (borjainglefernandez) [inline precommit.sh:13]: Also, we have to run each test individually for command line to run correctly.

File: tcf_website/templates/browse/browse.html [hunk 5]
```diff
@@ -15,7 +15,7 @@
         {% include "../common/leaderboard_ad.html" with ad_slot="9691204298" %}
 
         <!-- Browse page content -->
-        <h2 class="font-weight-light mt-4">Browse by Department</h2>
+        <h2 class="font-weight-light mt-4" id="browse-title">Browse by Department</h2>
```
Comment (borjainglefernandez) [inline tcf_website/templates/browse/browse.html:18]: Had to add id's to certain elements or else it was impossible for selenium to access them

**Comment by borjainglefernandez** [comment]
# Instructions:
    -After ./precommit.sh you can specify 1, 2, or no arguments
    -To have browser appear: ./precommit show
    -To specify different local host: ./precommit <local host>
    -To specify both: ./precommit show <url>
    -Default values: local host = "http://localhost:8000/", headless = True (order does not matter)
    Ex: ./precommit show http://localhost:8000/
Note that if you want headless mode on and you have http://localhost:8000/ as your local host, you do not have to specify arguments.

File: seleniumtests/utils_selenium_test.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        chrome_options.add_argument('--headless')
+
+    obj.driver = webdriver.Chrome(ChromeDriverManager().install(), options=chrome_options)
+
+def setup(obj):
+    """Navigate to url and get rid of welcome modal"""
+    obj.driver.get(obj.local_host)
+    thank_you_button = obj.driver.find_elements_by_id("thankYouButton")
+    if len(thank_you_button) == 1:
+        thank_you_button[0].click()
```
Comment (borjainglefernandez) [inline seleniumtests/utils_selenium_test.py:50]: Note to ad trailing white space

File: seleniumtests/browse_selenium_test.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
+"""Tests for Browse page."""
+
+import unittest
+
+from selenium.webdriver.support.ui import WebDriverWait
```
Comment (borjainglefernandez) [inline seleniumtests/browse_selenium_test.py:5]: I am not too familiar with the documentation style for imports so if this is wrong lmk

**Comment by alex-shen1** [comment]
Gonna put the "In Progress" label back on this one until Dockerization gets sorted out.

**Pull request closed without merging**